shell.nix

```diff
@@ -1,6 +1,12 @@
-{ packageSet ? import ./default.nix { rev = "in-nix-shell"; }
+{ packageSet ? import ./default.nix { rev = "in-nix-shell"; inherit sourcesOverride checkMaterialization useCabalProject; }
+, sourcesOverride ? {}
+, checkMaterialization ? false
```

Comment:
I find sourcesOverride useful at least.  checkMaterialization gives me some peace of mind, but is there a good update script I should use instead?

Reply:
I definitely don't object to checking the materialization, but we already have that as a `default.nix` argument and set it in CI, so I'm just not sure why I'd want to do it when I'm launching my shell. I guess you might worry that you'd missed something? (but we really want to make sure that the CI route gets everything, otherwise people will check in broken things)

But I get the argument for passing things through to `default.nix`. Maybe we should just forward the whole `args` to it?